packages/shared/src/retry.ts

```diff
@@ -0,0 +1,108 @@
+type Milliseconds = number;
+
+type RetryOptions = Partial<{
+  /**
+   * The initial delay before the first retry.
+   * @default 125
+   */
+  firstDelay: Milliseconds;
+  /**
+   * The maximum delay between retries.
+   * If set to 0, the delay will increase indefinitely.
+   * @default 0
+   */
+  maxDelay: Milliseconds;
+  /**
+   * The multiplier for the exponential backoff.
+   * @default 2
+   */
+  factor: number;
+  /**
+   * A function to determine if the operation should be retried.
+   * The callback accepts the error that was thrown and the number of iterations.
+   * The iterationsCount variable references the number of retries AFTER attempt
+   * that caused the error and starts at 1 (as in, this is the 1st, 2nd, nth retry).
+   * @default (error, iterationsCount) => iterationsCount < 5
+   */
+  shouldRetry: (error: unknown, iterationsCount: number) => boolean;
+  /**
+   * Controls whether the helper should retry the operation immediately once before applying exponential backoff.
+   * @default true
+   */
+  retryImmediatelyOnce: boolean;
+  /**
+   * The delay for the immediate retry.
+   * @default 100
+   */
+  retryImmediatelyDelay: Milliseconds;
+  /**
+   * If true, the intervals will be multiplied by a factor in the range of [1,2].
+   * @default true
+   */
+  jitter: boolean;
```

Comment:
Seems technical, how about `randomizeBackoff`?

Reply:
Not a fan of `randomizeBackoff` to be honest - `jitter` is the most common term for random intervals between retries in the context of handling transient errors (all AWS/MS guides etc)